File: HRForz-API/app/core/cache/keys.py
from uuid import UUID


class CacheKeys:
    """Single source of truth for all Redis key patterns."""

    @staticmethod
    def dropdown_departments() -> str:
        return "dropdown:departments"

    @staticmethod
    def dropdown_designations() -> str:
        return "dropdown:designations"

    @staticmethod
    def dropdown_work_locations() -> str:
        return "dropdown:work_locations"

    @staticmethod
    def dropdown_holiday_calendars() -> str:
        return "dropdown:holiday_calendars"

    @staticmethod
    def dropdown_salary_structures() -> str:
        return "dropdown:salary_structures"

    @staticmethod
    def employee_profile(employee_id: UUID) -> str:
        return f"employee:profile:{employee_id}"

    @staticmethod
    def attendance_today(employee_id: UUID) -> str:
        return f"attendance:today:{employee_id}"

    @staticmethod
    def leave_balance(employee_id: UUID) -> str:
        return f"leave:balance:{employee_id}"

    @staticmethod
    def refresh_token(user_id: UUID) -> str:
        return f"auth:refresh:{user_id}"

    @staticmethod
    def rate_limit(ip: str, endpoint: str) -> str:
        return f"ratelimit:{ip}:{endpoint}"

    @staticmethod
    def failed_login(ip: str) -> str:
        return f"auth:failed:{ip}"

File: HRForz-API-Node/src/core/cache/keys.ts
export class CacheKeys {
  static dropdownDepartments(): string {
    return 'dropdown:departments';
  }

  static dropdownDesignations(): string {
    return 'dropdown:designations';
  }

  static dropdownWorkLocations(): string {
    return 'dropdown:work_locations';
  }

  static dropdownHolidayCalendars(): string {
    return 'dropdown:holiday_calendars';
  }

  static dropdownSalaryStructures(): string {
    return 'dropdown:salary_structures';
  }

  static employeeProfile(employeeId: string): string {
    return `employee:profile:${employeeId}`;
  }

  static attendanceToday(employeeId: string): string {
    return `attendance:today:${employeeId}`;
  }

  static leaveBalance(employeeId: string): string {
    return `leave:balance:${employeeId}`;
  }

  static refreshToken(userId: string): string {
    return `auth:refresh:${userId}`;
  }

  static rateLimit(ip: string, endpoint: string): string {
    return `ratelimit:${ip}:${endpoint}`;
  }

  static failedLogin(ip: string): string {
    return `auth:failed:${ip}`;
  }
}
